riscv/triggers.cc

```diff
@@ -254,10 +308,26 @@ match_result_t module_t::memory_access_match(action_t * const action, operation_
   return MATCH_NONE;
 }
 
+match_result_t module_t::trap_taking_match(action_t * const action, trap_t& t)
```

Comment:
Can this be `const trap_t& t`?

Rather than modifying `*action`, can `action` be part of the `match_result_t`?